src/packages/backend/execute-code.ts

```diff
@@ -115,16 +151,90 @@ async function executeCodeNoAggregate(
       await writeFile(tempPath, cmd);
       await chmod(tempPath, 0o700);
     }
-    return await callback(doSpawn, { ...opts, origCommand });
-  } finally {
-    // clean up
-    if (tempDir) {
-      await rm(tempDir, { force: true, recursive: true });
+
+    if (opts.async_call) {
+      // we return an ID, the caller can then use it to query the status
+      opts.max_output ??= 1024 * 1024; // we limit how much we keep in memory, to avoid problems;
+      opts.timeout ??= 10 * 60;
+      const job_id = uuid();
+      const start = new Date();
+      const started: ExecuteCodeOutputAsync = {
+        type: "async",
+        stdout: `Process started running at ${start.toISOString()}`,
+        stderr: "",
+        exit_code: 0,
+        start: start.getTime(),
+        job_id,
+        status: "running",
+      };
+      asyncCache.set(job_id, started);
```

Comment:
jonathan: you could use a third arg to set the timeout here to be bigger than opts.timeout